inference-engine/thirdparty/clDNN/kernel_selector/core/common/kernel_selector_utils.cpp

```diff
@@ -199,20 +199,192 @@ std::vector<size_t> GetTensorFriendlyWorkGroups(const DataTensor& t) {
+std::vector<size_t> GetOptimalLocalWorkGroupSizes(std::vector<size_t> gws, const EngineInfo& info,
+                                                  DataLayout input_layout, DataLayout output_layout, std::vector<size_t> axis_by_dims) {
+    enum axis { x, y, z, w, f, b };
+
+    // GWS/LWS priority order should be considered for better local WGS setting
+    // and as a result more optimized data reading/writing inside kernels
+    std::vector<size_t> priority_order = { 0, 1, 2 };
+    std::vector<size_t> layout_order = { x, y, z, w, f, b };
+
+    const size_t gws_dims_num = 3;
+    const size_t axis_num = 6;
+    size_t first_axis_idx = 0;
+
+    auto calculate_optimized_priority_order = [&]() -> void {
+        while (axis_by_dims[layout_order[first_axis_idx]] == UNUSED_AXIS)
+            first_axis_idx++;
+
+        for (size_t gws_idx = 0; gws_idx < gws_dims_num; gws_idx++) {
+            for (size_t axis_idx = first_axis_idx; axis_idx < axis_num; axis_idx++) {
+                if (axis_by_dims[layout_order[axis_idx]] != UNUSED_AXIS) {
+                    bool is_already_exists = false;
+                    if (axis_idx > 0) {
+                        for (int i = axis_idx - 1; i >= 0; i--) {
+                            if (axis_by_dims[layout_order[axis_idx]] == axis_by_dims[layout_order[i]]) {
+                                is_already_exists = true;
+                                break;
+                            }
+                        }
+                    }
+                    first_axis_idx++;
+                    if (!is_already_exists) {
+                        priority_order[gws_idx] = axis_by_dims[layout_order[axis_idx]];
+                        break;
+                    }
+                }
+            }
+        }
+    };
+
+    auto one_layout = input_layout == output_layout;
+
+    auto simple_planar_layout = output_layout == DataLayout::bfyx || output_layout == DataLayout::yxfb ||
+                                output_layout == DataLayout::byxf || output_layout == DataLayout::fyxb ||
+                                output_layout == DataLayout::bfxy;
+
+    auto blocked_fsv_layout = output_layout == DataLayout::b_fs_yx_fsv4 || output_layout == DataLayout::fs_b_yx_fsv32 ||
+                              output_layout == DataLayout::b_fs_yx_fsv16 || output_layout == DataLayout::b_fs_zyx_fsv16 ||
+                              output_layout == DataLayout::b_fs_yx_fsv32 || output_layout == DataLayout::b_fs_zyx_fsv32;
+
+    auto blocked_bsv_fsv_layout = output_layout == DataLayout::bs_fs_yx_bsv16_fsv16 || output_layout == DataLayout::bs_fs_zyx_bsv16_fsv16;
+
+    auto try_change_priority_order = (simple_planar_layout || blocked_fsv_layout || blocked_bsv_fsv_layout) && one_layout;
+
+    if (try_change_priority_order) {
+        if (simple_planar_layout) {
+            switch (output_layout) {
+                case DataLayout::bfyx:
+                    layout_order = { x, y, f, b, z, w };
```

Comment:
Can we reuse order information from `dataChannelArray` structure?

Reply:
No, we can't due to another data format needed for `GetOptimalLocalWorkGroupSizes` function. Can be reused only for planar formats, but for blocked formats this solution won't work.